internal/corazarules/rule.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package corazarules
+
+import "github.com/corazawaf/coraza/v3/types"
+
+// RuleMetadata is used to store rule metadata
+// that can be used across packages
+type RuleMetadata struct {
+	ID_       int
+	File_     string
+	Line_     int
+	Rev_      string
+	Severity_ types.RuleSeverity
+	Version_  string
+	Tags_     []string
+	Maturity_ int
+	Accuracy_ int
+	Operator_ string
+	Phase_    types.RulePhase
+	Raw_      string
+	SecMark_  string
+}
+
+func (r *RuleMetadata) ID() int {
```

Comment:
I am curious about this methods. I see while we have them, code is still calling `.ID_` which I think could be private. Any particular reason for this?

Reply:
Many of the operators are technically internal details but mutate these fields. In a follow-up, I want to see if those can be moved to this package instead of the operators package since they're tied to implementation detail anyways